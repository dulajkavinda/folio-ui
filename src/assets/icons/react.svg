<svg width="34" height="30" viewBox="0 0 34 30" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M27.3361 9.5174C26.9849 9.39839 26.6307 9.28814 26.2739 9.18675C26.3338 8.94483 26.389 8.70585 26.438 8.47106C27.2421 4.61559 26.7164 1.50954 24.9209 0.48688C23.1993 -0.493508 20.3839 0.52877 17.5405 2.97299C17.2602 3.21431 16.9862 3.46259 16.7187 3.71758C16.5396 3.54802 16.3573 3.38175 16.172 3.21885C13.192 0.605418 10.2049 -0.496055 8.41134 0.529534C6.69134 1.51298 6.18203 4.43288 6.90597 8.0868C6.978 8.44868 7.05932 8.8087 7.14986 9.1665C6.72705 9.28517 6.31906 9.41147 5.9286 9.54593C2.43473 10.7491 0.203491 12.6347 0.203491 14.5905C0.203491 16.6105 2.59883 18.6365 6.23785 19.8651C6.53287 19.9643 6.83035 20.0562 7.13001 20.1407C7.03269 20.5261 6.94629 20.9141 6.87091 21.3042C6.18075 24.8947 6.7197 27.7456 8.43506 28.7228C10.2067 29.732 13.1804 28.6948 16.0757 26.1948C16.3105 25.9916 16.5401 25.7826 16.7643 25.568C17.0541 25.8441 17.3515 26.1122 17.6564 26.3719C20.4611 28.7557 23.231 29.7182 24.9447 28.7384C26.7147 27.7263 27.29 24.6636 26.5432 20.9375C26.4843 20.6456 26.4183 20.3552 26.3453 20.0664C26.5541 20.0053 26.7591 19.9424 26.9589 19.8769C30.742 18.639 33.2035 16.6374 33.2035 14.5905C33.2035 12.6275 30.9002 10.7293 27.3361 9.5174ZM26.5156 18.5553C26.3351 18.6143 26.15 18.6712 25.9613 18.7266C25.5436 17.4206 24.98 16.032 24.2903 14.5972C24.9485 13.1967 25.4904 11.8258 25.8963 10.5284C26.2339 10.6249 26.5615 10.7266 26.8773 10.8339C29.9311 11.8724 31.7938 13.4075 31.7938 14.5905C31.7938 15.8505 29.7821 17.4862 26.5156 18.5553ZM25.1603 21.208C25.4905 22.8555 25.5376 24.3452 25.3188 25.5098C25.1224 26.5563 24.7271 27.2539 24.2383 27.5334C23.1984 28.128 20.9744 27.3551 18.5759 25.3164C18.2923 25.0749 18.0156 24.8257 17.7461 24.569C18.676 23.5646 19.6053 22.3969 20.5123 21.1C22.1076 20.9602 23.6148 20.7315 24.9816 20.4195C25.0474 20.6809 25.107 20.9438 25.1603 21.208ZM11.4539 27.4306C10.4379 27.785 9.6286 27.7952 9.1394 27.5166C8.09848 26.9236 7.66574 24.6345 8.25601 21.564C8.32617 21.2011 8.40651 20.8403 8.49693 20.4818C9.84864 20.7772 11.3449 20.9896 12.9439 21.1177C13.8571 22.3867 14.8132 23.5531 15.7768 24.5745C15.572 24.7702 15.3624 24.9609 15.1481 25.1465C13.8677 26.2519 12.5847 27.0362 11.4539 27.4306ZM6.69392 18.5475C5.08478 18.0043 3.75589 17.2983 2.84504 16.5278C2.02648 15.8356 1.61334 15.1481 1.61334 14.5905C1.61334 13.4036 3.40475 11.8897 6.39267 10.8608C6.75528 10.736 7.13478 10.6182 7.52859 10.5077C7.9416 11.8344 8.48327 13.2216 9.13695 14.6243C8.47476 16.0478 7.92536 17.4572 7.50835 18.7995C7.23483 18.7221 6.96327 18.6381 6.69392 18.5475ZM8.28952 7.8193C7.66935 4.6888 8.08121 2.32734 9.11775 1.73465C10.2217 1.10338 12.6629 2.00343 15.236 4.2601C15.4038 4.40759 15.5689 4.55807 15.7312 4.71146C14.7723 5.72839 13.825 6.88614 12.92 8.14766C11.3679 8.28975 9.88229 8.51792 8.5164 8.82362C8.43224 8.49082 8.35659 8.15586 8.28952 7.8193ZM22.5243 11.2911C22.2016 10.7407 21.867 10.1972 21.5207 9.66103C22.5736 9.79255 23.5824 9.96711 24.5298 10.1805C24.2454 11.0809 23.8909 12.0222 23.4736 12.9873C23.1686 12.4158 22.8521 11.8502 22.5243 11.2911ZM16.7193 5.70662C17.3696 6.40244 18.0208 7.17924 18.6613 8.02212C17.3619 7.96151 16.0604 7.96125 14.761 8.02136C15.4022 7.18624 16.059 6.41046 16.7193 5.70662ZM10.8772 11.3007C10.5534 11.8553 10.2428 12.4173 9.94558 12.9862C9.53514 12.0245 9.18387 11.0789 8.89706 10.1657C9.83859 9.95769 10.8426 9.78759 11.8888 9.65848C11.5387 10.1979 11.2013 10.7456 10.8772 11.3007ZM11.9189 19.6215C10.838 19.5023 9.81887 19.341 8.87759 19.1386C9.16892 18.2091 9.52792 17.2434 9.94726 16.2611C10.2457 16.8304 10.5576 17.3927 10.8827 17.9476C11.2171 18.518 11.5634 19.0767 11.9189 19.6215ZM16.7583 23.5724C16.09 22.8604 15.4236 22.0726 14.7727 21.2254C15.4046 21.2499 16.0488 21.2623 16.7035 21.2623C17.3761 21.2623 18.041 21.2474 18.6955 21.2187C18.0529 22.0812 17.4041 22.87 16.7583 23.5724ZM23.4869 16.2109C23.928 17.204 24.2998 18.1647 24.595 19.0782C23.6383 19.2939 22.6052 19.4676 21.5167 19.5966C21.8655 19.0504 22.2033 18.4975 22.5298 17.9381C22.8618 17.3694 23.1809 16.7936 23.4869 16.2109ZM21.3087 17.2424C20.8088 18.1001 20.2808 18.9415 19.7256 19.7653C18.7198 19.8356 17.7118 19.8706 16.7035 19.8701C15.6765 19.8701 14.6776 19.8385 13.7157 19.7769C13.146 18.955 12.6079 18.1121 12.1027 17.25C11.5999 16.3925 11.1303 15.5164 10.6948 14.6237C11.1292 13.7308 11.5975 12.8543 12.0986 11.9961C12.5995 11.1377 13.1331 10.2986 13.6986 9.48023C14.6799 9.40689 15.6863 9.36869 16.7034 9.36869C17.7251 9.36869 18.7329 9.40727 19.714 9.48112C20.2737 10.2983 20.804 11.1347 21.304 11.9887C21.8068 12.845 22.2813 13.7172 22.7266 14.604C22.2858 15.4998 21.8129 16.3798 21.3087 17.2424ZM24.2168 1.69314C25.3218 2.32263 25.7515 4.86107 25.0573 8.18993C25.0129 8.40231 24.9632 8.61863 24.909 8.83801C23.5398 8.52581 22.0531 8.2937 20.4966 8.14944C19.5898 6.87405 18.6502 5.71452 17.707 4.7107C17.9542 4.47534 18.2073 4.24609 18.466 4.02315C20.9024 1.92907 23.1794 1.10223 24.2168 1.69314ZM16.7035 11.6799C18.3309 11.6799 19.6503 12.983 19.6503 14.5905C19.6503 16.1979 18.3309 17.5011 16.7035 17.5011C15.076 17.5011 13.7567 16.1979 13.7567 14.5905C13.7567 12.983 15.076 11.6799 16.7035 11.6799Z" fill="#00D8FF"/>
</svg>
