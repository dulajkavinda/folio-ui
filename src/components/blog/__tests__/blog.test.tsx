import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Blog from "../blog";

describe("components/blog", () => {
  it("should render blog", () => {
    const blog = renderer
      .create(
        <Blog
          title="title"
          description="description"
          category="category"
          date="date"
          link="link"
        />,
      )
      .toJSON();
    expect(blog).toMatchSnapshot();
  });

  it("should render with a given required text props", () => {
    render(
      <Blog
        title="title"
        description="description"
        category="category"
        date="date"
        link="link"
      />,
    );
    expect(screen.getByText("title")).toBeInTheDocument();
    expect(screen.getByText("description")).toBeInTheDocument();
    expect(screen.getByText("date")).toBeInTheDocument();
  });

  it("should render a Label with category", () => {
    render(
      <Blog
        title="title"
        description="description"
        category="category"
        date="date"
        link="link"
      />,
    );

    const blog = screen.getByTestId("folio-blog");

    expect(blog).toContainElement(screen.getByTestId("folio-label"));
    expect(screen.getByText("category")).toBeInTheDocument();
  });

  it("should not render description if not provided", () => {
    render(
      <Blog
        title="title"
        description=""
        category="category"
        date="date"
        link="link"
      />,
    );

    expect(screen.queryByTestId("folio-blog-description")).toBeNull();
  });

  it("should render description if provided", () => {
    render(
      <Blog
        title="title"
        description="description"
        category="category"
        date="date"
        link="link"
      />,
    );

    const blogDescription = screen.getByTestId("folio-blog-description");
    expect(blogDescription).toHaveClass("folio-blog_main__description");
  });

  it("should render a tech stack and a button if size is large if provided", () => {
    render(
      <Blog
        title="title"
        description="description"
        category="category"
        date="date"
        link="link"
        stack={["reactjs", "ts"]}
        size="large"
      />,
    );

    const blog = screen.getByTestId("folio-blog");
    expect(blog).toContainElement(screen.getByTestId("folio-tstack"));
    expect(blog).toContainElement(screen.getByTestId("folio-button"));
    expect(screen.getByText("Read More")).toBeInTheDocument();
  });

  it("should not render a tech stack if stack is not provided", () => {
    render(
      <Blog
        title="title"
        description="description"
        category="category"
        date="date"
        link="link"
        stack={[]}
        size="large"
      />,
    );

    expect(screen.queryByTestId("folio-tstack")).toBeNull();
  });

  it("should get classname and custom styles if provided", () => {
    render(
      <Blog
        title="title"
        description="description"
        category="category"
        date="date"
        link="link"
        classname="custom-class"
        customStyles={{ color: "red" }}
      />,
    );

    const blog = screen.getByTestId("folio-blog");
    expect(blog).toHaveClass("custom-class");
    expect(blog).toHaveStyle("color: red");
  });

  it("should set the provided link as href", () => {
    render(
      <Blog
        title="title"
        description="description"
        category="category"
        date="date"
        link="link"
      />,
    );

    const blog = screen.getByRole("link");
    expect(blog).toHaveAttribute("href", "link");
  });
});
