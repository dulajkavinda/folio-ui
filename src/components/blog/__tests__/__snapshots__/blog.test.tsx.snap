// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/blog should render blog 1`] = `
<a
  href="link"
  style={
    {
      "textDecoration": "none",
    }
  }
>
  <div
    className="folio-blog folio-blog--light folio-blog--small"
    data-testid="folio-blog"
    style={{}}
  >
    <div
      className="folio-blog_main"
    >
      <div
        className="folio-blog_main__title"
      >
        title
      </div>
      <div
        className="folio-blog_main__info"
      >
        <div
          className="folio-blog_main__date"
        >
          date
        </div>
        <div
          className="folio-label folio-label--light folio-label--medium"
          data-testid="folio-label"
          style={{}}
        >
          category
        </div>
      </div>
      <div
        className="folio-blog_main__description"
        data-testid="folio-blog-description"
      >
        description
      </div>
    </div>
  </div>
</a>
`;
